include/zephyr/drivers/dma.h

```diff
@@ -136,17 +136,25 @@ struct dma_block_config {
 	uint16_t  reserved :          3;
 };
 
+#define DMA_STATUS_FULL		0
+#define DMA_STATUS_BLOCK	1
+
 /**
  * @typedef dma_callback_t
  * @brief Callback function for DMA transfer completion
  *
- *  If enabled, callback function will be invoked at transfer completion
- *  or when error happens.
- *
- * @param dev Pointer to the DMA device calling the callback.
- * @param user_data A pointer to some user data or NULL
- * @param channel The channel number
- * @param status 0 on success, a negative errno otherwise
+ *  If enabled, callback function will be invoked at transfer or block completion,
+ *  or when an error happens.
+ *  In circular mode, @p status indicates that the DMA device has reached either
+ *  the end of the buffer (DMA_STATUS_FULL) or a water mark (DMA_STATUS_BLOCK).
+ *
+ * @param dev           Pointer to the DMA device calling the callback.
+ * @param user_data     A pointer to some user data or NULL
+ * @param channel       The channel number
+ * @param status        - 0-DMA_STATUS_FULL buffer fully consumed
```

Comment:
STATUS_BLOCK is probably fine and could easily relate to the block of data up to the water mark in the circular buffer cases and directly matches with the dma block config.

STATUS_FULL is sort of trying to lean too heavily on the circular buffer terms here when it could really mean the transfer list is complete for nxp edma or dw dma (atmel's xdmac works in a similar way).

STATUS_FULL just feels like the wrong terminology in those cases and needs a tweak

STATUS_DONE? STATUS_COMPLETE? something along those lines maybe.

Reply:
I'll change to `STATUS_COMPLETE`